WinUIB2CApp/MSAL/MSALClientHelper.cs

```diff
@@ -0,0 +1,248 @@
+            var storageProperties = new StorageCreationPropertiesBuilder(AzureADB2CConfig.CacheFileName, AzureADB2CConfig.CacheDir).Build();
+            var msalcachehelper = await MsalCacheHelper.CreateAsync(storageProperties);
+            msalcachehelper.RegisterCache(this.PublicClientApplication.UserTokenCache);
+
+            // If the cache file is being reused, we'd find some already-signed-in accounts
+            return await this.PublicClientApplication.GetAccountsAsync().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Signs in the user and obtains an Access token for a provided set of scopes
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <returns> Access Token</returns>
+        public async Task<string> SignInUserAndAcquireAccessToken(string[] scopes)
+        {
+            var existingUser = await FetchSignedInUserFromCache().ConfigureAwait(false);
+
+            try
+            {
+                // 1. Try to sign-in the previously signed-in account
+                if (existingUser != null)
+                {
+                    this.AuthResult = await this.PublicClientApplication.AcquireTokenSilent(scopes, existingUser)
+                        .ExecuteAsync().ConfigureAwait(false);
+                }
+                else
+                {
+
+                    this.AuthResult = await SignInUserInteractivelyAsync(scopes);
+                }
+            }
+            catch (MsalUiRequiredException ex)
+            {
+                // A MsalUiRequiredException happened on AcquireTokenSilentAsync. This indicates you need to call AcquireTokenInteractive to acquire a token interactively
+                Debug.WriteLine($"MsalUiRequiredException: {ex.Message}");
+
+                // Must be called from UI thread
+                this.AuthResult = await this.PublicClientApplication.AcquireTokenInteractive(scopes)
+                    .WithLoginHint(existingUser?.Username ?? String.Empty)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+            }
+            catch (MsalException msalEx)
+            {
+                Debug.WriteLine($"Error Acquiring Token:{Environment.NewLine}{msalEx}");
+            }
+
+            return this.AuthResult.AccessToken;
+        }
+
+        /// <summary>
+        /// Signs the in user and acquire access token for a provided set of scopes.
+        /// </summary>
+        /// <param name="scopes">The scopes.</param>
+    /// <param name="extraclaims">The extra claims, usually from CAE. We basically handle CAE by sending the user back to Azure AD for additional processing and requesting a new access token</param>
+        /// <returns></returns>
+        public async Task<String> SignInUserAndAcquireAccessToken(string[] scopes, string extraclaims)
+        {
+            var existingUser = await FetchSignedInUserFromCache().ConfigureAwait(false);
```

Comment:
Please keep the code [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). Merge this method with the one above.